packages/vitest/src/node/workspace/resolveWorkspace.ts

```diff
@@ -158,7 +158,7 @@ async function resolveWorkspaceProjectConfigs(
       const stringOption = definition.replace('<rootDir>', vitest.config.root)
       // if the string doesn't contain a glob, we can resolve it directly
       // ['./vitest.config.js']
-      if (!stringOption.includes('*')) {
+      if (!stringOption.includes('*') && !stringOption.startsWith('!')) {
```

Comment:
This should probably use `isDynamicPattern` from fast-glob https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#isdynamicpatternpattern-options, but it looks like it's not supported yet on tinyglobby https://github.com/SuperchupuDev/tinyglobby/issues/28

Reply:
Can we copy paste it maybe? Other cases includes {} for example

Reply:
Sounds good to me. I copied `isDynamicPattern` from fast-glob.